src/DevBetterWeb.Core/Services/NewMemberService.cs

```diff
@@ -0,0 +1,140 @@
+﻿using DevBetterWeb.Core.Entities;
+using DevBetterWeb.Core.Exceptions;
+using DevBetterWeb.Core.Interfaces;
+using DevBetterWeb.Core.Specs;
+using DevBetterWeb.Core.ValueObjects;
+using System;
+using System.Threading.Tasks;
+
+namespace DevBetterWeb.Core.Services
+{
+  public class NewMemberService : INewMemberService
+  {
+
+    private readonly IRepository _repository;
+    private readonly IUserRoleMembershipService _userRoleMembershipService;
+    private readonly IPaymentHandlerSubscription _paymentHandlerSubscription;
+    private readonly IEmailService _emailService;
+
+    public NewMemberService(IRepository repository,
+      IUserRoleMembershipService userRoleMembershipService,
+      IPaymentHandlerSubscription paymentHandlerSubscription,
+      IEmailService emailService)
+    {
+      _repository = repository;
+      _userRoleMembershipService = userRoleMembershipService;
+      _paymentHandlerSubscription = paymentHandlerSubscription;
+      _emailService = emailService;
+    }
+
+    public async Task<Invitation> CreateInvitation(string email, string stripeSubscriptionId)
+    {
+      var inviteCode = Guid.NewGuid().ToString();
+      var invitation = new Invitation(email, inviteCode, stripeSubscriptionId);
+
+      await _repository.AddAsync(invitation);
+
+      return invitation;
+    }
+
+    public async Task SendRegistrationEmail(Invitation invitation)
+    {
+      string code = invitation.InviteCode;
+      string inviteEmail = invitation.Email;
+
+      string completeRegistrationUrl = GetRegistrationUrl(code, inviteEmail);
+
+      // TODO: send Discord invite here as well
+
+      var message = "Thank you for joining DevBetter! Please click to complete your registration: " + completeRegistrationUrl + "\n";
+
+      await _emailService.SendEmailAsync(inviteEmail, "Welcome to DevBetter!", message);
+    }
+
+    public async Task<string> VerifyValidEmailAndInviteCode(string email, string inviteCode)
+    {
+      var spec = new InvitationByInviteCodeWithEmailSpec(inviteCode);
+
+      string ValidEmailAndInviteCode = "success";
```

Comment:
Think about replacing this string-based result with Ardalis.Result nuget package.